File: backend/src/main/kotlin/com/etchedjournal/etched/security/SecurityUser.java
package com.etchedjournal.etched.security;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;

/**
 * Implementation of Spring's UserDetails. This object is stored inside
 * getAuthentication().getPrincipal()
 */
public class SecurityUser implements UserDetails {

    private String username;
    private String password;
    private Collection<GrantedAuthority> authorities;
    private Long userId;

    public SecurityUser(String username, String password, Collection<GrantedAuthority>
            authorities, Long userId) {
        this.username = username;
        this.password = password;
        this.authorities = authorities;
        this.userId = userId;
    }

    public Long getUserId() {
        return userId;
    }

    @Override
    public String getUsername() {
        return username;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public Collection<GrantedAuthority> getAuthorities() {
        return authorities;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: backend/src/main/kotlin/com/etchedjournal/etched/security/SecurityUser.kt
package com.etchedjournal.etched.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

/**
 * Implementation of Spring's UserDetails. This object is stored inside
 * getAuthentication().getPrincipal()
 */
class SecurityUser(
        private val username: String,
        private val password: String,
        private val authorities: Collection<GrantedAuthority>,
        val userId: Long?) : UserDetails {

    override fun getUsername(): String {
        return username
    }

    override fun getPassword(): String {
        return password
    }

    override fun getAuthorities(): Collection<GrantedAuthority> {
        return authorities
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }
}
